crowd_anki/main.py

```diff
@@ -7,10 +7,28 @@
 
 from .anki.hook_vendor import HookVendor
 from .anki.ui.action_vendor import ActionVendor
+from .config.config_dialog import ConfigDialog
+from .config.config_settings import ConfigSettings
 
 
-def anki_actions_init(window):
-    action_vendor = ActionVendor(window, QAction, lambda caption: QFileDialog.getExistingDirectory(caption=caption))
+def invoke_config_window():
+    """
+    Launch custom GUI on config change instead of default Anki JSON editor
+    """
+    mw.crowd_anki_config = config = ConfigDialog()
```

Comment:
why do we need to inject a variable on the mw for it?

Reply:
It is suggested to do this in the [Writing Anki 2.1 Add-ons documentation](https://apps.ankiweb.net/docs/addons.html#qt), so that the garbage collection does not delete the window when it is not used.

Reply:
I see.

Reply:
in the current implementation though we'd recreate the widget from scratch each time anyhow (as we call a constructor here each time config action is invoked). So doing this does not really give us anything

Reply:
Changed to singleton :+1:

Reply:
err, this is a different discussion I believe ;)

Reply:
Ahh I understand your point now! Okay I have moved the line to the initialize_config_window function, and just left the invoke_config_window to execute it. Like this:

```

def invoke_config_window():
    """
    Launch custom GUI on config change instead of default Anki JSON editor
    """
    mw.crowd_anki_config.exec_()


def initialize_config_window():
    """
    Add option for addon's config in Anki
    :return:
    """
    mw.crowd_anki_config = ConfigDialog()
    mw.addonManager.setConfigAction(__name__, invoke_config_window)

```

 Tested and it works :+1: Does this sound better? :grin:

Reply:
this one is fine :) , creating a new widget each time was probably also fine tbh - it just didn't make sense to me to both create it each time and store the reference as advised by the link you give 😛